packages/flutter/lib/src/rendering/list_wheel_viewport.dart

```diff
@@ -529,35 +553,169 @@ class RenderListWheelViewport
+    final double crossPosition = size.width / 2.0 - child.size.width / 2.0;
+    childParentData.offset = new Offset(crossPosition, indexToScrollOffset(index));
+  }
+
+  /// Layout base on children provided by [childManager].
+  ///
+  /// From the current scroll offset, we can calculate the minimum index and
+  /// maximum index that is visible in the viewport. We can also get the index
+  /// range of the children that are currently active.
+  /// We will modify the child list so that children that are not visible
+  /// will be removed, and those that should visible but not yet created will be
+  /// instantiate by the childManager.
   @override
   void performLayout() {
-    double currentOffset = 0.0;
+    final BoxConstraints childConstraints =
+      constraints.copyWith(
+        minHeight: _itemExtent,
+        maxHeight: _itemExtent,
+        minWidth: 0.0,
+      );
+
+    // The height, in pixel, that children will be visible and might be laid out
+    // and painted.
+    double visibleHeight = size.height;
+    // If [renderChildrenOutsideViewport] is true, we spawn a redundant amount
+    // of children, those that are in the backside of the cylinder won't be
+    // painted anyway.
+    if (renderChildrenOutsideViewport)
+      visibleHeight *= 2;
+
+    final double firstVisibleOffset =
+        offset.pixels + _itemExtent / 2 - visibleHeight / 2;
+    final double lastVisibleOffset = firstVisibleOffset + visibleHeight;
+
+    // The index range that we want to spawn children. We find indexes that
+    // are in the interval [firstVisibleOffset, lastVisibleOffset).
+    int targetFirstIndex = scrollOffsetToIndex(firstVisibleOffset);
+    int targetLastIndex = scrollOffsetToIndex(lastVisibleOffset);
+    // Because we exclude lastVisibleOffset, if there's a new child starting at
+    // that offset, it is removed.
+    if (targetLastIndex * _itemExtent == lastVisibleOffset)
+      targetLastIndex--;
+
+    // Validate the target range.
+    while (!childManager.childExistsAt(targetFirstIndex) && targetFirstIndex <= targetLastIndex)
+      targetFirstIndex++;
+    while (!childManager.childExistsAt(targetLastIndex) && targetFirstIndex <= targetLastIndex)
+      targetLastIndex--;
+
+    // If it turns out there's no children to layout, we remove old children and
+    // return.
+    if (targetFirstIndex > targetLastIndex) {
+      while (firstChild != null)
+        _destroyChild(firstChild);
+      return;
+    }
+
+    // Now there are 2 cases:
+    //  - The target child list and our current child list have intersection:
+    //    We shorten and extend our current child list so that the two lists
+    //    match. Most of the time we are in this case.
+    //  - The target list and our current child list have no intersection:
+    //    We first remove all children and then add one child from the target
+    //    list => this case becomes the other case.
+
+    // Case when there is no intersection.
```

Comment:
I can't see at a glance where the code for the other case lives.

This is something you could consider splitting up into helper methods.

Reply:
The no-intersection case happens when the controller jumps to some position that is very far away (for example, when testing). It is true that we will never encounter the second case only by manual scrolling.
Can you be more specific about helper methods?